RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingSlotsContainer/mobileAppointmentBookingSlotsContainer.js

```diff
@@ -0,0 +1,549 @@
+    this.LABELS.Appointment_ReBooking_MonthName_April,
+    this.LABELS.Appointment_ReBooking_MonthName_May,
+    this.LABELS.Appointment_ReBooking_MonthName_June,
+    this.LABELS.Appointment_ReBooking_MonthName_July,
+    this.LABELS.Appointment_ReBooking_MonthName_August,
+    this.LABELS.Appointment_ReBooking_MonthName_September,
+    this.LABELS.Appointment_ReBooking_MonthName_October,
+    this.LABELS.Appointment_ReBooking_MonthName_November,
+    this.LABELS.Appointment_ReBooking_MonthName_December
+  ];
+
+  DAYNAME = [
+    this.LABELS.Appointment_ReBooking_WeekDayLong_Sunday,
+    this.LABELS.Appointment_ReBooking_WeekDayLong_Monday,
+    this.LABELS.Appointment_ReBooking_WeekDayLong_Tuesday,
+    this.LABELS.Appointment_ReBooking_WeekDayLong_Wednesday,
+    this.LABELS.Appointment_ReBooking_WeekDayLong_Thursday,
+    this.LABELS.Appointment_ReBooking_WeekDayLong_Friday,
+    this.LABELS.Appointment_ReBooking_WeekDayLong_Saturday
+  ];
+
+  @api get selecteddate() {
+    return this.selectedDate;
+  }
+  set selecteddate(value) {
+    if (value) {
+      this.selectedDate = this.getDateWithoutTime(Date.parse(value));
+      console.log("Selected Date in timeSlot : " + this.selectedDate);
+      this.firstDayOfTheWeek = this.getFirstDayOfWeek(this.selectedDate, 0);
+      this.lastDayOfTheWeek = this.getLastDayOfWeek(this.selectedDate, 0);
+
+      this.handleTimeSlotTitle();
+      this.handleDateSelectEvent(this.selectedDate);
+      this.afterDateSelection(this.selectedDate);
+      this.executeScroll(this.formatTitle(this.selectedDate));
+    }
+  }
+
+  @api get timeslotobject() {
+    return this.currentTimeSlotsObj;
+  }
+  set timeslotobject(value) {
+    let updatedData;
+    if (value) {
+      this.lockScrolling();
+      this.updateSlotData(value);
+      updatedData = Object.values(
+        this.slotsData[this.slotsData.currentAssignmentMethodRef]
+      );
+      this.handleTimeSlotUpdateEvent(updatedData);
+    } else {
+      this.formattedTimeSlotArray = [];
+      this.formattedRecommendedSlotsArray = [];
+    }
+  }
+
+  @api get showexactarrivaltime() {
+    return this.showExactArrivalTime;
+  }
+  set showexactarrivaltime(value) {
+    this.showExactArrivalTime = value;
+  }
+  isWeekUpdated = false;
+
+  @api onWeekUpdated(date) {
+    let slotsData;
+
+    this.isWeekUpdated = true;
+    console.log("Week change in parent class " + date);
+    this.firstDayOfTheWeek = this.getFirstDayOfWeek(date, 0);
+    this.lastDayOfTheWeek = this.getLastDayOfWeek(date, 0);
+    this.selectedDate = date;
+    slotsData = Object.values(
+      this.slotsData[this.slotsData.currentAssignmentMethodRef]
+    );
+    this.handleTimeSlotUpdateEvent(slotsData);
+  }
+
+  @api get shownoofdaysbeforeafterweek() {
+    return this.noOfDaysBeforeAfterWeek;
+  }
+  set shownoofdaysbeforeafterweek(value) {
+    if (value) {
+      this.noOfDaysBeforeAfterWeek = value;
+    }
+  }
+
+  @api get maxvaliddate() {
```

Comment:
The naming conventions of e.g. this getter is inconsistent with others in the file (like `onWeekUpdated()` above).

Reply:
Those are not the same exactly maxvaliddate and the others are props passed from parent component and onWeekUpdated is a public function called using this.template.querySelector.

Reply:
- All of the components—parents and children—are controlled in this body of source, so in theory the relationships can be changed to best practices.
- To that end, these properties don't fit the convention of property passing between LWC templates and the properties defined in their JS files. Public / `@api` property names between template and code should follow the convention of the property name in code being lower-camel-cased, and its reference in the template should use dash separators ([source](https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.js_props_names)).\
\
For example, `mobileAppointmentBookingLanding.html` should be referencing properties on `mobileAppointmentBookingSchedulingContainer` as follows:

        <c-mobile-appointment-booking-scheduling-container max-valid-date={maxValidCalendarDate} />

    Then you should define the naming of `maxvaliddate` as `maxValidDate` in `mobileAppointmentBookingSchedulingContainer`.